### PR TITLE
Add documentation to Array#pop

### PR DESCRIPTION
**Pull request by dylandrop**
Adds documentation to two undocumented variants of `Array#pop`.

### CODE REVIEW

File: src/array.cr [hunk 1]
```diff
@@ -1201,6 +1201,13 @@ class Array(T)
...
   end
 
+  # Removes the last value from `self`.
+  # If the array is of size 0, the given block is called.
+  #
+  # ```
+  # a = [1]
+  # a.pop { "Testing" } #=> 1
+  # a.pop { "Testing" } #=> "Testing"
   def pop
```
Comment (dylandrop) [inline src/array.cr]: Is this method even meant to be publicly exposed? It sure is an odd duck in terms of behavior. If it's meant to be private, perhaps it's best if it's hidden from the [API docs](https://crystal-lang.org/api/0.20.1/Array.html#pop%28%26block%29-instance-method) altogether.

File: src/array.cr [hunk 2]
```diff
@@ -1201,6 +1201,13 @@ class Array(T)
     pop { raise IndexError.new }
   end
 
+  # Removes the last value from `self`.
+  # If the array is of size 0, the given block is called.
```
Comment (asterite) [inline src/array.cr]: What about "If the array is empty"?

File: src/array.cr (hunk 1 above)
Comment (asterite) [inline src/array.cr]: Yes, it's public. Maybe you'd like to pop a value or use another default value instead of `nil`

**Comment by dylandrop** [comment]
@asterite Updated.

**Comment by asterite** [comment]
Thanks!

**Pull request merged**